generator/internal/discovery_resource.cc

```diff
@@ -138,21 +138,31 @@ StatusOr<std::string> DiscoveryResource::FormatRpcOptions(
   }
   rpc_options.push_back(absl::StrCat(http_option, "    };"));
 
-  std::vector<std::string> params =
+  std::vector<std::string> parameter_order =
       method_json.value("parameterOrder", std::vector<std::string>{});
-  if (!params.empty()) {
+  if (!parameter_order.empty()) {
+    // Workaround for necessary, but not marked REQUIRED, mask field for update
+    // methods.
+    if (verb == "patch") {
+      nlohmann::json parameters =
+          method_json.value("parameters", nlohmann::json());
+      if (parameters.find("updateMask") != parameters.end()) {
+        parameter_order.emplace_back("updateMask");
```

Comment:
I think we  should insert the `updateMask` only if it is not there already.  Someday the discovery doc may be fixed (or partially fixed) and we don't need the extra work.  Thoughts?